numpyro/distributions/kl.py

```diff
@@ -199,3 +200,30 @@ def kl_divergence(p, q):
     m = jnp.arange(1, p.KL_KUMARASWAMY_BETA_TAYLOR_ORDER + 1)
     t3 = (beta - 1) * b * (jnp.exp(betaln(m / a_, b_)) / (m + a_b_)).sum(-1)
     return t1 + t2 + t3
+
+
+@dispatch(MultivariateNormal, MultivariateNormal)
+def kl_divergence(p: MultivariateNormal, q: MultivariateNormal) -> jnp.DeviceArray:
+    """Compute the KL divergence from p to q for two multivariate normals.
+
+    Args:
+        p (MultivariateNormal): A multivariate normal distribution object.
+        q (MultivariateNormal): A second multivariate normal distribution object.
+
+    Returns:
+        jnp.DeviceArray: The corresponding KL divergence.
+    """
+    N = p.mean.shape[0]
+    iS1 = jnp.linalg.inv(q.covariance_matrix)
+    diff = q.mean - p.mean
+
+    # kl is made of three terms
+    tr_term = jnp.trace(jnp.matmul(iS1, p.covariance_matrix))
```

Comment:
We typically perform computation using `scale_tril` to improve speed, something like
```
from jax.scipy.linalg import solve_triangular
tmp = solve_triangular(q.scale_tril, p.scale_tril, lower=True)
tr_term = jnp.square(tmp.reshape(tmp.shape[:-2] + (-1,))).sum(-1)
```